hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/string/ConcatWSFunctionIntegrationTest.java

```diff
@@ -0,0 +1,214 @@
+import com.hazelcast.sql.HazelcastSqlException;
+import com.hazelcast.sql.SqlColumnType;
+import com.hazelcast.sql.impl.SqlDataSerializerHook;
+import com.hazelcast.sql.impl.expression.ConstantExpression;
+import com.hazelcast.sql.impl.expression.ExpressionTestSupport;
+import com.hazelcast.sql.support.expressions.ExpressionBiValue;
+import com.hazelcast.sql.support.expressions.ExpressionType;
+import com.hazelcast.sql.support.expressions.ExpressionTypes;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static com.hazelcast.sql.impl.type.QueryDataType.INT;
+import static com.hazelcast.sql.impl.type.QueryDataType.VARCHAR;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class ConcatWSFunctionIntegrationTest extends ExpressionTestSupport {
+
+    @Test
+    public void testColumn() {
+        for (ExpressionType<?> type1 : ExpressionTypes.all()) {
+            // Test supported types
+            for (ExpressionType<?> type2 : ExpressionTypes.all()) {
+                Class<? extends ExpressionBiValue> clazz = ExpressionBiValue.createBiClass(type1.typeName(), type2.typeName());
+
+                checkColumns(
+                        ExpressionBiValue.createBiValue(clazz, 0, type1.valueFrom(), type2.valueFrom()),
+                        type1.valueFrom() + "-" + type2.valueFrom()
+                );
+
+                checkColumns(
+                        ExpressionBiValue.createBiValue(clazz, 0, null, type2.valueFrom()),
+                        type2.valueFrom().toString()
+                );
+
+                checkColumns(
+                        ExpressionBiValue.createBiValue(clazz, 0, type1.valueFrom(), null),
+                        type1.valueFrom().toString()
+                );
+
+                checkColumns(
+                        ExpressionBiValue.createBiValue(clazz, 0, null, null),
+                        ""
+                );
+            }
+        }
+    }
+
+    @Test
+    public void testLiteral() {
+        put("1");
+
+        check(getConcatWsExpression("-", "this", "2"), "1-2");
+        check(getConcatWsExpression("-", "this", "'2'"), "1-2");
+        check(getConcatWsExpression("-", "this", "2e0"), "1-2E0");
+
+        check(getConcatWsExpression("-", "this", "true"), "1-true");
+
+        check(getConcatWsExpression("-", "this", "null"), "1");
+
+        check(getConcatWsExpression("-", "1", "2"), "1-2");
+        check(getConcatWsExpression("-", "'1'", "2"), "1-2");
+    }
+
+    @Test
+    public void testNull() {
+        put("1");
+
+        //Null on separator => returns null
+        checkFail(getConcatWsExpression("null", false, "3", "2"));
```

Comment:
Not sure if this should fail. Postgres simply returns `null` 🤔

Reply:
Yeah that was the intended behaviour, the `checkFail` method was working impaired it turns out, now its fixed